src/spatialRegistry/splitTree/Node.java

```diff
@@ -69,7 +69,44 @@ public void add( Entry<T> entry )
 			{
 				this.getEntries().add(entry);
 				if( this.size() > this._tree.maxEntries &! this._atomic )
-					split();
+				{
+					/*
+					 * This boolean tells you whether the entry
+					 * has a higher high and a lower low than the
+					 * node. Checking this ensures that you do not
+					 * get stuck in an endless loop of creating
+					 * smaller and smaller nodes
+					 */
+					boolean entryEncompassesNode = true;
+					
+					for (int i = 0; i < entry.getLow().length; i++)
+					{
+						if (entry.getLow()[i] > this.getLow()[i]
+								|| entry.getHigh()[i] < this.getHigh()[i])
+							entryEncompassesNode = false;
+					}
+					
+					/*
+					 * Checks whether the entry's range in the
```

Comment:
The already existing backstop for this is !this._atomic

This would only happen if there is something seriously wrong with your simulation, I've only seen this when there is a high amount of overlapping agents in a single area. If this happens in your simulation then you have to adjust mechanical relaxation settings or reduce your stepsizes. The only other case in which this would happen is using a random agent spawner with a lot of agents, but this should be resolved quickly in the first iterations of the mechanical relaxation (and !this_atomic will effectively prevent an endless loop).

I would suggest reverting the changes in Node unless there is some other reason why this is required.